specs/01-sharded-archiving.md

```diff
@@ -0,0 +1,288 @@
+  triggers the proof of the shard's segment in the global history of the beacon chain.
+- To verify the `SegmentInfo` and execute the transaction, nodes verify the following:
+  - Ensure `segment_index` is the subsequent index after the last one committed for the shard.
+- If the verification of the `SegmentInfo` is successful, the child shard segment is added to the
+  beacon chain shard's history by adding the shard's `segment_root` to the `segment_roots[]` that
+  indexes the all proofs from the global history (including beacon chain and all child segments).
+- Apart from adding the segment proof in the global history, a new `SuperSegment` is created
+  aggregating all the segment proofs included in a block of the beacon chain. The goal for this
+  super segment proof is to simplify the verification of segments by light clients.
+- The `super_segment` is created by aggregating the segment proofs of the child segments and
+  creating a new proof that is included in the parent shard's history committing (and wrapping) all
+  the history of shards in the lower layers of the architecture. This `super_segment` proof is
+  directly a Merkle proof of all of the shard segments and child super segment proofs from that
+  shard: `C_super_segment = Merkle_Proof(S1, ..., Sn, SS1, ..., SSn)`.
+- Along with `C_super_segment` the corresponding `ws1` segments proof are created.
+- When a new super segment is created instead of broadcasting the network the full super segment,
+  only the `SuperSegmentHeader` is broadcasted, allowing nodes to proactively determine if they
+  require the full super segment data.
+- The `SuperSegmentHeader` includes the following fields:
+  - `history_delta`: Number of segment proofs included in the `super_segment`. This will be used to
+    determine the number of segments committed in the beacon chain at a specific `block_height`.
+  - `super_segment_root` proof of the super segment that implicitly verifies into the global history
+    all the included segments.
+  - `block_height` in the beacon chain where the super segment was created. This is used to identify
+    the block in the child shard that contains the segments included in the super segment and pull
+    all the super segment data (if needed).
+- Output: Add segment proofs from children to `segment_roots[]`, and a new `SuperSegment` per block
+  with segment proofs.
+
+```rust
+type SuperSegmentHeader = {
+	super_segment_root: Vec<u8>,
+  block_height: u64,
+	history_delta: u64,
+};
+```
+
+#### 9. Recursive Local Block and Segment Creation and Submission
+
+- Input: Transactions with `BlockInfo`, new local segments, and `SegmentInfo`.
+- The protocol is recursive, which means that if we have a hierarchical architecture with more than
+  one level below the beacon chain, all shards independently of the level they belong to will
+  perform the same operations described in the sections from above: i.e. submit new
```

Comment:
This is not exactly accurate. Super segments according to our latest discussion are only created on the beacon chain, so at least that is different.